src/cookieSyncManager.ts

```diff
@@ -0,0 +1,184 @@
+
+            // Kit Module ID
+            const moduleId = pixelSettings.moduleId.toString();
+
+            // Tells you how often we should do a cookie sync (in days)
+            const frequencyCap = pixelSettings.frequencyCap;
+
+            // Url for cookie sync pixel
+            const pixelUrl = replaceAmpWithAmpersand(pixelSettings.pixelUrl);
+
+            // TODO: Document requirements for redirectUrl
+            const redirectUrl = pixelSettings.redirectUrl
+                ? replaceAmpWithAmpersand(pixelSettings.redirectUrl)
+                : null;
+
+            const urlWithRedirect = this.combineUrlWithRedirect(
+                mpid,
+                pixelUrl,
+                redirectUrl
+            );
+
+            // TODO: Is there a historic reason for checking for previousMPID?
+            //       it does not appear to be passed in anywhere
+            if (previousMPID && previousMPID !== mpid) {
+                if (persistence && persistence[mpid]) {
+                    if (!persistence[mpid].csd) {
+                        persistence[mpid].csd = {};
+                    }
+                    self.performCookieSync(
+                        urlWithRedirect,
+                        moduleId,
+                        mpid,
+                        persistence[mpid].csd,
+                        filteringConsentRuleValues,
+                        mpidIsNotInCookies,
+                        requiresConsent
+                    );
+                }
+                return;
+            } else {
+                if (!persistence || !persistence[mpid]) {
+                    return;
+                }
+
+                if (!persistence[mpid].csd) {
+                    persistence[mpid].csd = {};
+                }
+
+                const lastSyncDateForModule = persistence[mpid].csd[moduleId]
+                    ? persistence[mpid].csd[moduleId]
+                    : null;
+
+                // Check to see if we need to refresh cookieSync
+                if (hasFrequencyCapExpired(lastSyncDateForModule, frequencyCap)) {
+                    self.performCookieSync(
+                        urlWithRedirect,
+                        moduleId,
+                        mpid,
+                        persistence[mpid].csd,
+                        filteringConsentRuleValues,
+                        mpidIsNotInCookies,
+                        requiresConsent
+                    );
+                }
+
+            }
+        });
+    };
+
+    this.combineUrlWithRedirect = (
+        mpid: MPID,
+        pixelUrl: string,
+        redirectUrl
+    ): string => {
+        const url = replaceMPID(pixelUrl, mpid);
+        const redirect = redirectUrl ? replaceMPID(redirectUrl, mpid) : '';
+        return url + encodeURIComponent(redirect);
+    };
+
+    // Private
+    this.performCookieSync = (
+        url: string,
+        moduleId: string,
+        mpid: MPID,
+        cookieSyncDates: Dictionary<number>,
```

Comment:
we use `cookieSyncDates: Dictionary<number>` in enough places that i think you should just make it its own Type